frontend/catalyst/ag_utils.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions for Catalyst's AutoGraph module. This module can be safely imported without
+a TensorFlow installation."""
+
+import inspect
+
+import pennylane as qml
+
+# pylint: disable=import-outside-toplevel
+
+
+class AutoGraphError(Exception):
+    """Errors related to Catalyst's AutoGraph module."""
+
+
+def _test_ag_import():
+    """Reusable function for attempting to import Catalyst's AutoGraph module, which requires
+    TensorFlow to be installed."""
+
+    try:
+        import catalyst.autograph  # pylint: disable=unused-import
+    except ImportError as e:
+        raise ImportError(
+            "The AutoGraph feature in Catalyst requires TensorFlow. "
+            "Please install it (e.g. `pip install tensorflow-cpu`) and make it sure is "
+            "available in the current environment."
+        ) from e
+
+
+def converted_code(fn):
+    """Utility function to retrieve the source code of a function converted by AutoGraph.
+
+    Args:
+        fn (Callable): the original function object that was converted
+
+    Returns:
+        str: the source code of the converted function
+
+    Raises:
+        AutoGraphError: If the given function was not converted by AutoGraph, an error will be
+                        raised.
+        ImportError: If TensorFlow is not installed, an error will be raised.
+
+    **Example**
+
+    .. code-block:: python
+
+        def decide(x):
+            if x < 5:
+                y = 15
+            else:
+                y = 1
+            return y
+
+        @qjit(autograph=True)
+        def func(x):
+            y = decide(x)
+            return y ** 2
+
+        print(converted_code(decide))
```

Comment:
```suggestion
    >>> print(converted_code(decide))
    'example'
```

Reply:
don't forget to include the output here! I think that would be really nice in demonstrating to someone how cool this new feature is, if they chance upon it in the docs.

Reply:
Sounds good, didn't include it originally because the output is not that pretty/decipherable for someone unfamiliar with the inner workings of autograph (it's basically just dispatching code, not the code we eventually want to run).

Reply:
I'm not sure the `>>>` syntax will work well here because of the large multi-line output the code produces.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/235/commits/5faa3e9ac123fef4871b3b4852dcb64740e1aded